src/renderer.ts

```diff
@@ -153,6 +156,430 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+        path.enter().append('path');
+        path
+          .attr('class', 'boxplotline')
+          .attr('d', function (d, i) {
+            const left = boxdata.min, right = boxdata.max, center = boxdata.median;
+            const bottom = Math.max(context.rowHeight(i) - padding, 0);
+            const middle = (bottom - padding) / 2;
+
+            return 'M' + left + ',' + middle + 'L' + boxdata.q1 + ',' + middle +
+              'M' + left + ',' + padding + 'L' + left + ',' + bottom +
+              'M' + center + ',' + padding + 'L' + center + ',' + bottom +
+              'M' + (boxdata.q3) + ',' + middle + 'L' + (right) + ',' + middle +
+              'M' + right + ',' + padding + 'L' + right + ',' + bottom;
+
+          });
+        path.exit().remove();
+
+      }
+    };
+  }
+
+  createCanvas(col: Column, context: ICanvasRenderContext): ICanvasCellRenderer {
+    const multiValueColumn: MultiValueColumn = (<MultiValueColumn>col);
+
+    const padding = context.option('rowPadding', 1);
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      // Rectangle
+      const boxdata = multiValueColumn.getboxPlotData(col.getValue(d.v, d.dataIndex));
+      ctx.fillStyle = '#e0e0e0';
+      ctx.strokeStyle = 'black';
+      ctx.beginPath();
+      ctx.rect(boxdata.q1, padding, (boxdata.q3 - boxdata.q1), context.rowHeight(i) - padding);
+      ctx.fill();
+      ctx.stroke();
+
+      //Line
+      const left = boxdata.min, right = boxdata.max, center = boxdata.median;
+      const bottom = Math.max(context.rowHeight(i) - padding, 0);
+      const middle = (bottom - padding) / 2;
+      ctx.strokeStyle = 'black';
+      ctx.fillStyle = '#e0e0e0';
+      ctx.beginPath();
+      ctx.moveTo(left, middle);
+      ctx.lineTo(boxdata.q1, middle);
+      ctx.moveTo(left, padding);
+      ctx.lineTo(left, bottom);
+      ctx.moveTo(center, padding);
+      ctx.lineTo(center, bottom);
+      ctx.moveTo(boxdata.q3, middle);
+      ctx.lineTo(right, middle);
+      ctx.moveTo(right, padding);
+      ctx.lineTo(right, bottom);
+      ctx.stroke();
+      ctx.fill();
+
+    };
+  }
+
+
+}
+
+class UpsetCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: Column, context: IDOMRenderContext): ISVGCellRenderer {
+    const upsetColumn: UpsetColumn = (<UpsetColumn>col);
+    const celldimension = upsetColumn.cellDimension();
+
+    return {
+      template: `<g class="upsetcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const circle = d3.select(n).selectAll('circle').data(<any>col.getValue(d.v, d.dataIndex));
+        circle.enter().append('circle');
+        circle
+          .attr({
+            cy: (d: any, i) => (context.rowHeight(i) / 2),
+            cx: (d: any, i) => (i * celldimension) + (celldimension / 2),
+            r: (celldimension / 4),
+            class: 'upsetcircle',
+            opacity: (d) =>(d === 1) ? 1 : 0.1
```

Comment:
the renderer should make the same assumption that `1` means part of or use at least a constant from the model